### PR TITLE
refactor(interface): Use default export for implementation

### LINKED ISSUE #59: Can implementation classes default-export implementation, instead of having a single export?
We don't have any potential exports beside `implementation`, right? This would be a decent amount of churn on the jsdom side, but it seems easier to explain.

### PR DESCRIPTION
**Pull request by ExE-Boss**
# ⚠️ **WARNING:** BREAKING CHANGE

Fixes #59

---

This changes the implementation class syntax to allow CommonJS and ES2015 style default exports:
```js
// CommonJS
/** @type {WeakMap<Foo, PrivateData>} */
const fooPrivateData = new WeakMap();

class Foo {
	// implementation
}
module.exports = exports = Foo;

export.init = function(instance, privateData) {
	fooPrivateData.set(instance, privateData);
}
```

```js
// ES2015
/** @type {WeakMap<Foo, PrivateData>} */
const fooPrivateData = new WeakMap();

export default class Foo {
	// implementation
}

export function init(instance, privateData) {
	fooPrivateData.set(instance, privateData);
}
```

---

## TODO:
- [x] Add tests for the `importStar(…)` helper.
- [x] Update documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,20 +15,33 @@ interface SomeInterface {
...
 ```js
-exports.implementation = class SomeInterfaceImpl {
+module.exports = class SomeInterfaceImpl {
   add(x, y) {
     return x + y;
   }
 };
 ```
 
+> Note: It's also possible to use ES2015 module default export syntax:
```
Comment (domenic) [inline README.md]: This is not true; jsdom only operates on CommonJS modules.

File: lib/output/utils.js [hunk 2]
```diff
@@ -74,6 +74,52 @@ const namedSetNew = Symbol("named property set new");
 const namedSetExisting = Symbol("named property set existing");
 const namedDelete = Symbol("named property delete");
 
+/**
```
Comment (domenic) [inline lib/output/utils.js]: I am not interested in this addition. Please remove it.

Comment (ExE-Boss) [reply lib/output/utils.js]: Which? The&nbsp;JSDoc or&nbsp;the&nbsp;entire `utils.importStar(…)`&nbsp;function?

File: README.md (hunk 1 above)
Comment (ExE-Boss) [reply README.md]: Well, you need to transpile them first.

File: lib/output/utils.js (hunk 2 above)
Comment (domenic) [reply lib/output/utils.js]: The importStar function itself.

File: README.md (hunk 1 above)
Comment (domenic) [reply README.md]: Yes. That is not a supported scenario, so please remove it.

**Pull request closed without merging**